### DISCUSSION TITLE
Cache Size and setup in self-hosted

### DISCUSSION THREAD

**Discussion by danez** [renovatebot/renovate#9161] [Request Help] [answered]
We use renovate in kubernetes and have a persistent volume for renovate to write its cache.
Currently we have 
```
baseDir: "/renovate-cache/",
```
`persistRepoData` is no set.

The persisted volume is 15GiB which lasts for about 3 to 4 weeks before it is full. Out of this problem, we increased it now to 50GiB, but there are some questions that arose:

- What type of data is stored in `baseDir` vs `cacheDir`?
- Is it expected that one of them or both be ever-growing? or would they level out at some point?
- Is there some cleanup happening or should we do manual cleanup jobs?
- Is there any recommendation if we should only persist `cacheDir` or `baseDir`?

Thanks.

**Comment by HonkingGoose** [comment]
@rarkins do you have any help/advice?

**Comment by rarkins** [comment, marked as answer]
> * What type of data is stored in `baseDir` vs `cacheDir`?

`cacheDir` is a subdirectory of `baseDir` unless configured explicitly. The cache includes the package cache (version lookups, changelogs, etc) as well as package manager caches.

> * Is it expected that one of them or both be ever-growing? or would they level out at some point?

I think the majority of size comes from the package manager caches, and those are not cleaned up periodically. As long as there's new releases/modules, they'd get added to the cache, although I would have expected the growth to slow down a lot.

> * Is there some cleanup happening or should we do manual cleanup jobs?

There's no automatic cleanup, and you can do a manual cleanup any time Renovate is not running - it's totally OK to remove the cache between runs.

> * Is there any recommendation if we should only persist `cacheDir` or `baseDir`?

When you are self hosting, I think that persisting `cacheDir` is generally a good idea due to the amount of network traffic you save yourself. Depends which is worse for you - using 30GB for cache or an additional terabyte of downloads a year or something like that?

**Reply by danez** [reply]
Thank you very much.
